tests/test_spectral_nomination.py

```diff
@@ -13,156 +19,94 @@
-        _nominate(adj, np.zeros(3, dtype=int), nominator=svn)
-
-
-class TestSpectralVertexNominatorOutputs:
     def test_seed_inputs(self):
-        _test_seed_input_dimensions()
-        _test_seed_shape()
-        _test_seed_input_array_dtype()
-        _test_seed_input_type()
+        with self.assertRaises(IndexError):
+            self._nominate(adj, np.zeros((5, 5, 5), dtype=np.int))
+        with self.assertRaises(IndexError):
+            self._nominate(adj, np.zeros((1, 50), dtype=np.int))
+        with self.assertRaises(TypeError):
+            self._nominate(adj, np.random.random((10, 2)))
+        with self.assertRaises(TypeError):
+            self._nominate(adj, [0] * 10)
 
     def test_X_inputs(self):
-        _test_X_array_dtype()
-        _test_X_input_dimensions()
-        _test_X_input_type()
-        _test_embedding_dimensions()
-        _test_adjacency_shape()
+        with self.assertRaises(TypeError):
+            self._nominate(np.random.random((10, 10)), np.zeros(3, dtype=np.int))
+        with self.assertRaises(IndexError):
+            self._nominate(np.zeros((5, 5, 5), dtype=np.int), np.zeros(3, dtype=np.int))
+        with self.assertRaises(TypeError):
+            self._nominate([[0] * 10] * 10, np.zeros(3, dtype=np.int))
+        # embedding should have fewer cols than rows.
+        svn = SpectralVertexNomination(input_graph=False)
+        with self.assertRaises(IndexError):
+            self._nominate(
+                np.zeros((10, 20), dtype=np.int),
+                np.zeros(3, dtype=np.int),
+                nominator=svn,
+            )
+        # adj matrix should be square
+        with self.assertRaises(IndexError):
+            self._nominate(np.zeros((3, 4), dtype=np.int), np.zeros(3, dtype=np.int))
 
     def _test_k(self):
-        _test_k_value()
-        _test_k_type()
+        # k should be > 0
+        with self.assertRaises(ValueError):
+            self._nominate(adj, np.zeros(3, dtype=np.int), k=0)
+        # k of wrong type
+        with self.assertRaises(TypeError):
+            self._nominate(adj, np.zeros(3, dtype=np.int), k="hello world")
 
     def test_constructor_inputs(self):
-        _test_embedder_value()
+        with self.assertRaises(ValueError):
+            svn = SpectralVertexNomination(embedder="hi")
+            self._nominate(adj, np.zeros(3, dtype=int), nominator=svn)
 
     def test_constructor_inputs1(self):
-        _test_embedder_type()
+        # embedder must be BaseSpectralEmbed or str
+        with self.assertRaises(TypeError):
+            svn = SpectralVertexNomination(embedder=45)
 
     def test_constructor_inputs2(self):
-        _test_input_graph_bool_type()
+        # input graph param has wrong type
+        with self.assertRaises(TypeError):
+            svn = SpectralVertexNomination(input_graph=4)
 
-    @pytest.mark.parametrize(
-        "nominator",
-        [
+    def test_basic_unattributed(self):
```

Comment:
This is an example of me taking some of the niceties of pytest and dumbing it down so that unittest is happy.  I'm hopeful that we can use pytest marks inside `unittest` tests, maybe as inner functions or something, but my brain felt scratched clean after making all of these updates and I couldn't manage it today.